interface/client/lib/helpers/helperFunctions.js

```diff
@@ -50,6 +50,28 @@ Helpers.formatUrl = function(url){
 };
 
 /**
+Sanatizes URLs to prevent phishing and XSS attacks
+
+@method sanatizeUrl
+@param {String} url
+**/
+Helpers.sanatizeUrl = function(url){
+    var parser;
+
+    if(_.isString(url)) {
+        parser = document.createElement('a');
+        parser.href = url;
+    } else if(_.isObject(url) && url.protocol) {
+        parser = url;
+    }
+
+    parser.protocol = parser.protocol.replace(/^(javascript:)?(data:)?(:)?/i, 'http:');
```

Comment:
What specifically did not work with the DOM-anchor method ? 
I like that you now force it to be a string, but dislike that it's gone back to regexp. URL parsing is by no means trivial, so I'd still advocate a proper URL parsing library if the DOM is insufficient.